src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -438,12 +440,32 @@ public static TestPackage MakeTestPackage(ConsoleOptions options)
             if (options.DefaultTestNamePattern != null)
                 package.AddSetting(FrameworkPackageSettings.DefaultTestNamePattern, options.DefaultTestNamePattern);
 
-            if (options.TestParameters != null)
-                package.AddSetting(FrameworkPackageSettings.TestParameters, options.TestParameters);
+            if (options.TestParameters.Count != 0)
+                AddTestParametersSetting(package, options.TestParameters);
 
             return package;
         }
 
+        /// <summary>
+        /// Sets test parameters, handling backwards compatibility.
+        /// </summary>
+        private static void AddTestParametersSetting(TestPackage testPackage, IDictionary<string, string> testParameters)
+        {
+            testPackage.AddSetting(FrameworkPackageSettings.TestParametersDictionary, testParameters);
+
+            if (testParameters.Count != 0)
+            {
+                // This cannot be changed without breaking backwards compatibility with old frameworks.
+                // Reserializes the way old frameworks understand, even if this runner's parsing is changed.
+
+                var oldFrameworkSerializedParameters = new StringBuilder();
+                foreach (var parameter in testParameters)
+                    oldFrameworkSerializedParameters.Append(parameter.Key).Append('=').Append(parameter.Value).Append(';');
+
+                testPackage.AddSetting(FrameworkPackageSettings.TestParameters, oldFrameworkSerializedParameters.ToString(0, oldFrameworkSerializedParameters.Length - 1));
```

Comment:
I don't think we do.

[@CharliePoole:](https://github.com/nunit/nunit/issues/1885#issuecomment-268686434)

> That said, I don't particularly want to guarantee order among the differently named parameters. It makes no sense to me. I do understand the desire for a particular parameter name to reference an array of values in a certain order, but that's another (future) thing.
>
> Unless you have a good reason of your own to use IEnumerable<KeyValuePair<string, string>>, I'd go with <IDictionary<string, string>> for the test parameters. That's what they end up as in TestContext anyway.

I agree with that quote. I feel somewhat strongly that the parameter names should be the unit of isolation between tests. If you need ordering, it means you're actually taking what should be a single parameter and spreading it across multiple parameters.